terrat_runner/workflow_step_terrateam_ssh_key_setup.py

```diff
@@ -0,0 +1,39 @@
+import logging
+import os
+import workflow
+import subprocess
+
+def run(state, config):
+    logging.info('TERRATEAM_SSH_KEY : SETUP')
+    env = state.env
+
+    # Create the ssh directory
+    ssh_dir_path = "/root/.ssh/"
+    if not os.path.exists(ssh_dir_path):
+        os.makedirs(ssh_dir_path)
+
+    # Iterate through environment variables that start with TERRATEAM_SSH_KEY
+    ssh_keys = [(key, val) for key, val in env.items() if key.startswith('TERRATEAM_SSH_KEY')]
+    for ssh_key_name, ssh_key_value in ssh_keys:
+        ssh_key_path = os.path.join(ssh_dir_path, ssh_key_name)
+        # Write the ssh key to a file
+        with open(ssh_key_path, 'w') as f:
+            f.write(ssh_key_value)
+            f.write('\n')
+        os.chmod(ssh_key_path, 0o600)
+        # Add the ssh key to the ssh-agent
+        subprocess.run(['ssh-add', '{}'.format(ssh_key_path)])
```

Comment:
`subprocess.run(['ssh-add', ssh_key_path])`

Also use `check_call` unless it's OK if this call fails.